c/src/core/object/object.c

```diff
@@ -58,50 +67,130 @@ pn_cid_t pn_class_id(const pn_class_t *clazz)
   return clazz->cid;
 }
 
+static inline void *pni_object_new(const pn_class_t *clazz, size_t size)
+{
+  void *object = NULL;
+  pni_head_t *head = (pni_head_t *) pni_mem_zallocate(clazz, sizeof(pni_head_t) + size);
+  if (head != NULL) {
+    object = head + 1;
+    head->clazz = clazz;
+    head->refcount = 1;
+  }
+  return object;
+}
+
+static inline void pni_object_incref(void *object) {
+  if (object) {
```

Comment:
Should this assert instead of null check?  _refcount and _decref are (_decref needs an explicit assert, I think).  Maybe the null check in pn_object_incref, not pni_object_incref?

In general, I'd prefer all the refcounting methods required a non-null object, but that does require fixing up a number of places.

Reply:
pn_class_incref/decref are actually allowed to be called with a NULL object so they really do need to check for null and do nothing if they get that - the other inlined functions may not need the check.